src/jit/PerfDump.cpp

```diff
@@ -0,0 +1,175 @@
+    JIT_CODE_LOAD = 0, // record describing a jitted function
+    JIT_CODE_MOVE = 1, // record describing an already jitted function which is moved (optional)
+    JIT_DEBUG_INFO = 2, // record describing the debug information for a jitted function (optional)
+    JIT_CODE_CLOSE = 3, // record marking the end of the jit runtime (optional)
+    JIT_CODE_UNWINDING_INFO = 4 // record describing a function unwinding information
+};
+
+void PerfDump::openFile()
+{
+    std::string tmpName = "./jit-" + std::to_string(getpid()) + ".dump";
+    this->file = fopen(tmpName.c_str(), "w");
+    this->dumpFileHeader();
+
+    int fd = open(tmpName.c_str(), O_RDONLY | O_CLOEXEC, 0);
+    mmap(NULL, 1, PROT_READ | PROT_EXEC, MAP_SHARED, fd, 0);
+    close(fd);
+}
+
+void PerfDump::dumpFileHeader()
+{
+    struct {
+        const uint32_t magic;
+        const uint32_t version;
+        const uint32_t totalSize;
+        const uint32_t elfMach;
+        const uint32_t pad;
+        const uint32_t pid;
+        const uint64_t timestamp;
+        const uint64_t flags;
+    } fileHeader = {
+        .magic = DUMP_MAGIC,
+        .version = DUMP_VERSION,
+        .totalSize = sizeof(uint32_t) * 6 + sizeof(uint64_t) * 2,
+        .elfMach = ELFMACH,
+        .pad = 0,
+        .pid = (uint32_t)getpid(),
+        .timestamp = (uint64_t)std::time(0),
+        .flags = 0
+    };
+
+    fwrite(&fileHeader, sizeof(fileHeader), 1, file);
+}
+
+void PerfDump::dumpRecordHeader(const uint32_t recordType, const uint32_t entrySize)
+{
+    struct {
+        const uint32_t recordType;
+        const uint32_t size;
+        const uint64_t timestamp;
+    } recordHeader = {
+        .recordType = recordType,
+        .size = entrySize + ((uint32_t)sizeof(uint32_t)) * 2 + (uint32_t)sizeof(uint64_t),
+        .timestamp = (uint64_t)std::time(0)
+    };
+
+    fwrite(&recordHeader, sizeof(recordHeader), 1, file);
+}
+
+void PerfDump::dumpCodeLoad(const uint64_t vma, const uint64_t codeAddr, const uint64_t codeSize, const std::string& functionName, const uint8_t* nativeCode)
+{
+    struct {
+        const uint32_t pid;
+        const uint32_t tid;
+        const uint64_t vma;
+        const uint64_t codeAddr;
+        const uint64_t codeSize;
+        const uint64_t codeLoadIndex;
+    } codeLoad1 = {
+        .pid = pid,
+        .tid = (uint32_t)std::hash<std::thread::id>{}(std::this_thread::get_id()),
+        .vma = vma,
+        .codeAddr = codeAddr,
+        .codeSize = codeSize,
+        .codeLoadIndex = codeLoadIndex++
+    };
+    const size_t nameSize = functionName.size() + 1;
+
+    dumpRecordHeader(JIT_CODE_LOAD, sizeof(codeLoad1) + nameSize + codeSize);
+    fwrite(&codeLoad1, sizeof(codeLoad1), 1, file);
+    fwrite(functionName.c_str(), nameSize, 1, file);
+    fwrite(nativeCode, codeSize, 1, file);
+}
+
+void PerfDump::dumpCodeMove(const uint64_t vma, const uint64_t oldCodeAddr, const uint64_t newCodeAddr, const uint64_t codeSize, const uint64_t codeIndex)
```

Comment:
What is this function? Would you explain more about this?
It seems that this function is defined but not used at all.
This is also same for  `dumpCodeClose` function.

Reply:
https://elixir.bootlin.com/linux/latest/source/tools/perf/Documentation/jitdump-specification.txt

At some point we should support more events. Move is probably unnecessary, and close is optional. However, JIT_DEBUG_INFO could be useful, if we would generate a byte code dump (.txt), and it would point to lines there.